pkg/issuer/acme/dns/route53/route53.go

```diff
@@ -159,14 +169,6 @@ func (d *sessionProvider) GetSession(ctx context.Context) (aws.Config, error) {
 		)
 	}
 
-	// If ambient credentials aren't permitted, always set the region, even if to
-	// empty string, to avoid it falling back on the environment.
-	// This has to be set after session is constructed, as a different region (aws-global)
-	// is used for the STS service.
-	if d.Region != "" || !useAmbientCredentials {
-		cfg.Region = d.Region
-	}
-
```

Comment:
Since we merged https://github.com/cert-manager/cert-manager/pull/7287, the region field can be omitted.
So this code may have resulted in a situation where the region had been discovered from AWS_REGION, during `LoadDefaultConfig`.
But if `--issuer-ambient-credentials=False`, that would be overridden with an empty string, causing the Route53 operations to fail.